core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -2721,6 +2734,47 @@ private ExpressionNode nextLiteral(CharSequence token) {
         return nextLiteral(token, 0);
     }
 
+    private boolean nonAggregateFunctionDependsOn(ExpressionNode node, CharSequence argument) {
+        sqlNodeStack.clear();
+        while (node != null) {
+            if (node.type != FUNCTION) {
+                if (node.type == LITERAL && Chars.equalsIgnoreCase(node.token, argument)) {
+                    return true;
+                }
+            }
+
+            if (node.type != FUNCTION || !functionParser.getFunctionFactoryCache().isGroupBy(node.token)) {
+                if (node.paramCount < 3) {
+                    if (node.lhs != null) {
+                        sqlNodeStack.push(node.lhs);
+                    }
+
+                    if (node.rhs != null) {
+                        node = node.rhs;
+                        continue;
+                    }
+                } else {
+                    for (int i = 1, k = node.paramCount; i < k; i++) {
+                        if (node.args.getQuick(i) != null) {
```

Comment:
more for my enlightenment: can this ever be `null`? and if so, can `node.args.getQuick(0)` bellow be `null` too? if so, is it possible we exit this method with a non-empty stack and thus reporting possibly false-negative?

Reply:
It shouldn't be null, we can remove the if.